corehq/ex-submodules/casexml/apps/phone/restore.py

```diff
@@ -455,15 +514,19 @@ def get_cached_payload(self):
         else:
             payload = self.cache.get(self._initial_cache_key())
 
-        if payload and payload.endswith(FileRestoreResponse.EXTENSION) and not path.exists(payload):
-            return
-        return payload
+        return CachedResponse(payload)
 
     def set_cached_payload_if_necessary(self, resp, duration):
+        cache_payload = resp.get_cache_payload(bool(self.sync_log))
         if self.sync_log:
             # if there is a sync token, always cache
             try:
-                self.sync_log.set_cached_payload(resp, self.version)
+                data = cache_payload['data']
```

Comment:
Yea, this stores the actual data as an attachment on the SyncLog object whereas the redis cache value is a dict with either the full data or just the filename.